### DISCUSSION TITLE
Background Image Support

### DISCUSSION THREAD

**Discussion by ming900518** [zed-industries/zed#11086] [Feature Requests]
### Problem
In VSCode and IntelliJ, there's an option or extension to set a background image, which is missing in Zed.

<img width="890" alt="1" src="https://github.com/zed-industries/zed/assets/15919723/3d0e7c53-6b12-42e1-878d-1c127aeaa8e0">

### Solution Proposal
I forked this project and added an option called `background.image_file` to the `experimental.theme_overrides` setting.

Repo: https://github.com/ming900518/zed/tree/background-image

```json
{
  "experimental.theme_overrides": {
    "background.image_file": "*FILE PATH*",
  }
}
```

With this option set, Zed will read the image file specified and render it behind the UI elements, users can combine this new option with other background color options to create beautiful blending.

<img width="1440" alt="2" src="https://github.com/zed-industries/zed/assets/15919723/b78f7d41-ba1f-42cd-9e6c-53b714ef502c">

I would like to see this feature officially supported by Zed, so I looked up the [CONTRIBUTING.md](https://github.com/zed-industries/zed/blob/main/CONTRIBUTING.md) and opened this discussion to propose adding this feature.

I'm not quite sure if I should open the pull request directly or wait for the Zed team to respond with this proposal. If Zed team is interested and the pull request is preferred or required for proposing changes. Please let me know.

**Comment by martin503** [comment]
any update on this one?

**Reply by jqiris** [reply]
更功能加了吗？

**Comment by arjunindia** [comment]
+1 on this

**Comment by vipexv** [comment]
+1, need this :o

**Comment by trindade99** [comment]
+1 that's very useful! I'm wondering if theres no opacity option? I would like to see my background

**Comment by 0x2CA** [comment]
Background images need more features such as random display, timed switch, custom layout styles, fixed-size bottom-right alignment, transparency settings, and more.

**Comment by mjovanc** [comment]
This is awesome. Any updates on this?

**Comment by gatzer455** [comment]
+1

**Comment by JonGretar** [comment]
Even cooler would be to have a multi layer parallax available like in WezTerm. 😄
https://wezterm.org/config/lua/config/background.html

**Comment by ArtrenH** [comment]
+1

**Comment by ArekX** [comment]
+1 on this. This is pretty much one the BIG things keeping me from Zed.

**Comment by Elue-dev** [comment]
is this a thing now?

**Comment by Pependuez** [comment]
+1

**Comment by Klassikcat** [comment]
Any updates in here? And one more thing.

IntelliJ and VS Code cannot change the background image based on theme changes. I think it would be great if Zed had this option.

**Comment by jjtimi888** [comment]
Any update here?

**Comment by alperenyigiter** [comment]
Something new?

**Comment by frtian** [comment]
@ming900518 did you applied what @0x2CA said? Also I think you should add this as a PR directly as u said u've already made a fork and implemented it, FR are more like _"Hey can you guys add this?"_ instead of _"I've made this feature, can u guys apply it?"_

**Reply by MariusKnobbe** [reply]
They discourage opening PR's directly in their feature request guidelines.

**Comment by Aoimina** [comment]
If this get added im ditching vscode entirely, (i hope it support gifs too)
